Add patch to remove unused Qt dependency from vtkPVVTKExtensionsDefaults

martyngigg opened the pull request:
`vtkPVVTKExtensionsDefaults` depends on `vtkPVVTKExtensionsRendering`, which in turn depends on `vtkGUISupportQt` but from `ldd --unused` this library is not used at all. It does however cause issues in our combined Qt4/Qt5 build.

This additional patch removes the dependency from the library.

I tested this on RHEL7 by rebuilding ParaView with this patch and running all of the Vates unit tests.

quantumsteve commented:
What about cherry-picking the upstream commit?

https://github.com/Kitware/ParaView/commit/ab8cd653944c094b2e3d5e502072eaf112d35526#diff-5faa5d1b16fdf9f1b9753b140a366f44

martyngigg commented:
Good plan. I hadn't think to look if it had already been patched elsewhere.

martyngigg committed:
Add patch to remove unused Qt dependency from vtkPVVTKExtensionsDefaults

Required to be able to load VATES algorithms without a Qt dependency.
The patch is already in ParaView's master branch but cherry picking the
commit fails as there have been additional changes between our version
and that commit.

martyngigg commented:
Unfortunately the cherry-pick fails due to other changes to those 2 files between our version and the current `master` so I think we'll have to resort to a patch. It can be removed when we move to the next version of ParaView though.

Edit: I've updated the patch to include the other file that the `cherry-pick` would have affected.

martyngigg commented:
The RHEL 7 build and tests now pass on mantidproject/mantid#21297 after I rebuilt paraview on a single machine. Shall I go ahead and rebuild the rest?
